### ISSUE TITLE
Is this complete or not?

### ISSUE THREAD

**Issue by yuroc0598** [mpillar/java-emotion-recognizer#3]

**Comment by daleysoftware** [comment]
Hi @gabrielchen9527. Can you elaborate on what you mean by "complete"? :)

While I haven't worked on this project in many years, I can say that at the time of writing this code was perfectly functional and performed quite well.

If you have more specific questions about the project, I'd be happy to help.

**Comment by yuroc0598** [comment]
Hi Matt,

I'm so sorry for my mistake and I have to admit that your program is
wonderful and perfect. I just didn't understand some details of your codes
at the time I posted it as incomplete  and now I want to apologize to you.

I'm strongly interesting in your program of emotion recognition and may
have one or two questions about its details. May I ask you these technique
details through email?

(questions like in src/image/FacialImage.java, line 166~169, why does minX=
featAverage.getWidth()-1 while maxX=0 and the same for Y? Will minX > maxX
in this case? Because I got the error "can't find the face" )

Any way, I'm so grateful that you could reply and help. Thanks man!

Sincerely,

Gabriel

**Comment by daleysoftware** [comment]
Hey @gabrielchen9527, I'll answer this question here. If you have additional questions after this, feel free to shoot me an email.

The purpose of the algorithm in FacialImage.java around lines 166-189 is to identify a bounding box for the face in the given image. What I do here is essentially look at the red channel pixel by pixel. If the channel is above a certain threshold, we consider that to be part of the face, and adjust the bounding box accordingly.

This technique is quite obviously not bullet proof, so it's not surprising that at least in some cases the algo can't "find the face". Without seeing the samples you're using, one tip that comes to mind is brightening; try brightening the image, and perhaps that will help.

**Comment by yuroc0598** [comment]
Hey Matt, thanks! It might be the issue of my training image when getting
this error. Despite of this, I still don't understand why the maxX and maxY
are initially set to 0 while minX and minY are set to featAverage.getWidth()-1.
To me it seems that minX (or minY) will be larger than maxX (or maxY). Will
this cause an error later when determine the region of face where minX
should be less than maxX?

Thanks! I hope I'm not bothering you with this.

**Comment by yuroc0598** [comment]
and the image I used for training is CK+ (
http://www.pitt.edu/~emotion/ck-spread.htm). could you recommend the image
dataset that is suitable for your algorithm?

**Comment by daleysoftware** [comment]
Unfortunately I can't recommend a particular dataset; when I was working on this it was part of a school project, and we took all of the pictures ourselves! All I can say is play around with it; crop the images to limit the size of the background, adjust color, etc.

Regarding the confusion around minX, maxX, etc., the initialization there is intentional. Essentially I'm using their relative values to determine whether we were able to find a bounding box at all. In retrospect, a more clear "found" condition might have been helpful here for readability.
